vm/src/builtins/int.rs

```diff
@@ -751,6 +755,78 @@ impl Hashable for PyInt {
     }
 }
 
+impl AsNumber for PyInt {
+    const AS_NUMBER: PyNumberMethods = PyNumberMethods {
+        add: Some(|number, other, vm| Self::np_int_op(number, other, |a, b| a + b, vm)),
+        subtract: Some(|number, other, vm| Self::np_int_op(number, other, |a, b| a - b, vm)),
+        multiply: Some(|number, other, vm| Self::np_int_op(number, other, |a, b| a * b, vm)),
+        remainder: Some(|number, other, vm| Self::np_general_op(number, other, inner_mod, vm)),
+        divmod: Some(|number, other, vm| Self::np_general_op(number, other, inner_divmod, vm)),
+        power: Some(|number, other, vm| Self::np_general_op(number, other, inner_pow, vm)),
+        negative: Some(|number, vm| {
+            Self::number_downcast(number)
+                .value
+                .clone()
+                .neg()
+                .to_pyresult(vm)
+        }),
+        positive: Some(|number, vm| Ok(Self::np_int(number, vm).into())),
+        absolute: Some(|number, vm| Self::number_downcast(number).value.abs().to_pyresult(vm)),
+        boolean: Some(|number, _vm| Ok(Self::number_downcast(number).value.is_zero())),
+        invert: Some(|number, vm| {
+            let value = Self::number_downcast(number).value.clone();
```

Comment:
`Not` also implemented for reference.

https://docs.rs/num/latest/num/struct.BigInt.html#impl-Not-1